universal-application-tool-0.0.1/app/export/CsvExporter.java

```diff
@@ -1,9 +1,44 @@
 package export;
 
+import java.io.IOException;
 import java.io.Writer;
+import java.util.List;
+import java.util.Optional;
 import models.Applicant;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVPrinter;
 
 public class CsvExporter implements Exporter {
+  private boolean wroteHeaders;
+  private List<String> headers;
+  private List<String> columns;
+
+  public CsvExporter(List<String> headers, List<String> columns) {
+    this.wroteHeaders = false;
+    this.headers = headers;
+    this.columns = columns;
+  }
+
+  private void maybeWriteHeaders(CSVPrinter printer) throws IOException {
```

Comment:
nit: recommend renaming to `writeHeadersIfUnwriten` or somesuch

Reply:
The downside seems that the consumer needs to read the method body to know when it will write. If the logic changes I'd expect the method name to change -- something which should be especially easy given it's a private method.

Reply:
Done - `writeHeadersOnFirstExport`